### ISSUE TITLE
How to do cross-modal retrieval?

### ISSUE THREAD

**Issue by carlthome** [tensorflow/similarity#223]
I'm curious about how to do cross-modal retrieval. I have videos with image and audio data, and would like to learn two encoders that embed both audio and RGB data into the same space, such that nearest neighbor lookups could be performed with audio embeddings to find related videos, and vice versa.

Is there an easy way to extend the loss functions required by `SimilarityModel` to support two input heads?

`(features, labels) = ({'rgb': ..., 'audio': ...}, {'video_id': ...})`

**Comment by owenvallis** [comment]
This would similar to the [CLIP](https://github.com/openai/CLIP) model. We are looking to add an example notebook for this at some point.

**Comment by Layhan** [comment]
Hi i was wondering if you did 'Import CLIP'?
